File: C/练习/code_12_1.c
// 判断找出101到200之间的素数
#include <stdio.h>

void is_prime(void) {
    int i = 101, j = 200;
    int count = 0;
    for (i; i <= j; i++) {
        int mean = (int)(i / 2) + 1;
        int sign;
        for (int sub = 2; sub < mean; sub++) {
            if (i % sub == 0) {
                sign += 1;
                break;
            }else {
                sign = 0;
            }
        }
        if (sign == 0 ) {
            printf("数字 %d 是素数\n", i);
            count++;
        }else {
            continue;
        }
    }
    printf("在101到200之间共有 %d 个素数\n", count);
}

int main(void) {
    is_prime();
    return 0;
}

File: C/Replicate_With_Python/code_12_1.py
def is_prime(start, end):
    count = 0
    for i in range(start, end + 1):
        mean = (i // 2) + 1
        sign = 0
        for j in range(2, mean):
            if i % j == 0:
                sign += 1
                break
            else:
                continue
        if sign == 0:
            print(i, "是素数")
            count += 1
        if sign == 1:
            continue
    print(f"在{start}到{end}之间共有 {count} 个素数")
    return

is_prime(101, 200)
